File: app/src/main/java/com/example/buildmypc/ui/parts/parts/Accessory.java
package com.example.buildmypc.ui.parts.parts;

import android.os.Parcel;
import android.os.Parcelable;

import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;

public class Accessory extends Part implements Parcelable {
	public static final Creator<Accessory> CREATOR = new Creator<Accessory>() {
		@NotNull
		@Contract("_ -> new")
		@Override
		public Accessory createFromParcel(Parcel in) {
			return new Accessory(in);
		}

		@NotNull
		@Contract(value = "_ -> new", pure = true)
		@Override
		public Accessory[] newArray(int size) {
			return new Accessory[size];
		}
	};

	public Accessory(String model, String manufacturer) {
		super(model, manufacturer);
	}

	protected Accessory(Parcel in) {
		super(in);
	}

	@Override
	public void writeToParcel(@NotNull Parcel dest, int flags) {
		super.writeToParcel(dest, flags);
	}

	@Override
	public int describeContents() {
		return 0;
	}
}

File: app/src/main/java/com/example/buildmypc/ui/parts/parts/Accessory.kt
package com.example.buildmypc.ui.parts.parts

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator

open class Accessory : Part, Parcelable {
	constructor(model: String?, manufacturer: String?) : super(model, manufacturer)
	protected constructor(`in`: Parcel?) : super(`in`!!)

	override fun writeToParcel(dest: Parcel, flags: Int) = super.writeToParcel(dest, flags)

	override fun describeContents() = 0

	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (javaClass != other?.javaClass) return false
		if (!super.equals(other)) return false
		return true
	}

	override fun hashCode() = super.hashCode()

	companion object {
		@JvmField
		val CREATOR = object : Creator<Accessory> {
			override fun createFromParcel(`in`: Parcel) = Accessory(`in`)

			override fun newArray(size: Int): Array<Accessory?> = arrayOfNulls(size)
		}
	}
}